base/compiler/abstractinterpretation.jl

```diff
@@ -154,6 +161,26 @@ function abstract_call_gf_by_type(interp::AbstractInterpreter, @nospecialize(f),
         seen += 1
         rettype = tmerge(rettype, this_rt)
         rettype === Any && break
+        if this_conditional !== Bottom && is_lattice_bool(rettype) && !method.isva && fargs !== nothing
```

Comment:
The `t !== Bottom && t ⊑ Bool` pattern is now wrapped in `is_lattice_bool` check.